app/models/comment.php

```diff
@@ -20,17 +20,116 @@ public static function getComments($offset, $limit, $id)
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
+        foreach ($rows as $row) {
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));
+            $total_favorites = Comment::countFavorites($row['id']);
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
+        }
+        return $comments;
```

Comment:
Always add checker if an array is not empty before returning the value. Like this po:
```
        if (!$row) {
            throw new RecordNotFoundException('no record found');   
        }

        return new self($row);
```
:bow:  (Apply to all others)